src/electron/sslibev_badvpn_tunnel.ts

```diff
@@ -248,8 +269,8 @@ export class TunnelManager {
     this.tun2socks.stop();
   }
 
-  // Use #onceStopped to be notified when the tunnel terminates.
-  stop() {
+  // Use #onceDisconnected to be notified when the tunnel terminates.
+  async disconnect() {
     powerMonitor.removeListener('suspend', this.suspendListener.bind(this));
```

Comment:
There's an asymmetry here: you setup the listeners in the constructor, but remove on disconnect. You should probably add them on connect instead.

If you disconnect, then connect again, I guess you won't get the listeners.

Reply:
Done in aa2fd60.